utils/upload.go

```diff
@@ -177,8 +177,8 @@ func (a AzureUploader) Upload(walker fileWalk) {
 		_, err = azblob.UploadFileToBlockBlob(ctx, file, blockBlobUrl, azblob.UploadToBlockBlobOptions{})
 		if err != nil {
 			//close the file before error log
-			file.Close()
-			log.Println("Failure to upload to azure container:")
+			defer file.Close()
```

Comment:
using `defer` here will not pass the DeepSource checks. So I had removed `defer file.Close()` after `os.Open()` and used `file.Close()` every time an error which could stop the process occurred, and at the end after completion of the upload

Reply:
got it